Assets/Scripts/Utils/Probability.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System;
+
+namespace Utils
+{
+    /// <summary>
+    /// Utility class capable of dealing with small probabilities.
+    /// </summary>
+    public class Probability
+    {
+        private Random _random;
+
+        public Probability() => _random = new Random();
+
+        /// <summary>
+        /// Returns true with a certain probability.
+        /// </summary>
+        /// <param name="probability">The probability between 0 and 1 to return true</param>
+        /// <returns></returns>
+        public bool GetProbability(float probability)
+        {
+            if (probability < 0 || probability > 1f)
+                throw new Exception("Wrong usage of Probability");
+            if (probability == 0)
+                return false;
+            if (probability == 1)
+                return true;
+
+            float percentage = probability * 100;
+            int temp = _random.Next(100);
+            if (percentage >= 1)
+            {
+                return temp < percentage;
+            }
+            else // < 1% -----> new min: 0.01%
+            {
+                if (temp == 0) // hit the one percent
+                {
+                    // 1 old percent = 100 new percent
+                    return GetProbability(percentage);
```

Comment:
Won't this throw an exception every time? It takes the percentage value (which is between 0 and 100) and calls the same method again. Also, I'm not sure if you should call this recursive like this. Maybe you should calculate the value you multiply the probability with, with the probability you get? E.g. 0.01 - 100 / 0.001 - 1000? You could also use floats for the random as well I think

Reply:
It doesn't throw an exception because it's in the else that is only entered if percentage < 1 (e.g. 0.8).

> calculate the value you multiply the probability with, with the probability you get

I'm not exactly sure how you mean it, because the method simply doesn't return `true` with the correct probability if you multiply the passed probability with a value depending on the dimension of the probability. 😬 I like the recursion because it's not too difficult to read (and it works 😁 ). 
If you have a better solution, you can implement it of course.

Reply:
it's fine like that:)